packages/grid/x-data-grid/src/hooks/features/scroll/useGridScroll.ts

```diff
@@ -27,6 +27,11 @@ function scrollIntoView(dimensions: {
   const { clientHeight, scrollTop, offsetHeight, offsetTop } = dimensions;
 
   const elementBottom = offsetTop + offsetHeight;
+  // Avoid scroll jump when cell is higher/wider than viewport and is already visible
+  // See https://github.com/mui/mui-x/issues/4513 and https://github.com/mui/mui-x/issues/4514
+  if (offsetHeight > clientHeight && scrollTop === offsetTop) {
```

Comment:
I don't think only `scrollTop === offsetTop` will be sufficient. If I scroll a little bit it stills jumps to the right. I would expect that it would jump to the left to display the full cell. What do you think?

![msedge_RXkpyjr5eo](https://user-images.githubusercontent.com/42154031/163831140-52666827-ab7d-48e0-8e3b-42c44225d621.gif)

Reply:
I've played with it a bit more and ended up with always scrolling to top (left) when the cell is higher (wider) than viewport. Should be better now